kafka-connector/src/main/java/com/google/pubsub/kafka/sink/CloudPubSubSinkTask.java

```diff
@@ -317,20 +276,24 @@ private void publishMessagesForPartition(
       outstandingFutures = new OutstandingFuturesForPartition();
       outstandingFuturesForTopic.put(partition, outstandingFutures);
     }
-    int startIndex = 0;
-    int endIndex = Math.min(CPS_MAX_MESSAGES_PER_REQUEST, messages.size());
-    PublishRequest.Builder builder = PublishRequest.newBuilder();
-    // Publish all the messages for this partition in batches.
-    while (startIndex < messages.size()) {
-      PublishRequest request =
-          builder.setTopic(cpsTopic).addAllMessages(messages.subList(startIndex, endIndex)).build();
-      builder.clear();
-      log.trace("Publishing: " + (endIndex - startIndex) + " messages");
-      outstandingFutures.futures.add(publisher.publish(request));
-      startIndex = endIndex;
-      endIndex = Math.min(endIndex + CPS_MAX_MESSAGES_PER_REQUEST, messages.size());
+    outstandingFutures.futures.add(publisher.publish(message));
```

Comment:
Unless this topic must always match the topic created by the publisher, I think you may need to have the publishers be Map<String,Publisher> since we don't support multiple topics in the same publisher by default right now.

Reply:
The CPS topic is statically set in the config of the connector and used across all Kafka topics from which messages are received, so there is only a single topic to which we publish.

Reply:
Should we remove topic from being a parameter to this function then?

Reply:
No, topic refers to the _Kafka_ topic. The flush method is called for a specific Kafka topic and partition. When this method is called, we need to ensure that all messages for that specific Kafka topic and partition have successfully been published. This is why we track futures by topic and partition. The topic argument is not related to the CPS topic at all.